From: Hannes Reinecke <hare@suse.de>
Date: Wed, 18 Jul 2018 08:00:07 +0200
Subject: [PATCH] nvme: kabi fixes for nvme_ctrl
References: bsc#1054245
Patch-Mainline: never, kABI fix

The patch 'nvme-use-the-changed-namespaces-list-log-to-clear-ns.patch'
introduced a new field 'events' and 'oaes' in the nvme_ctrl structure,
which needs to be guarded to avoid kABI breakage.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/nvme.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/nvme/host/nvme.h b/drivers/nvme/host/nvme.h
index a5c1e28905d2..54480281d67f 100644
--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -180,7 +180,6 @@ struct nvme_ctrl {
 	u16 kas;
 	u8 npss;
 	u8 apsta;
-	u32 oaes;
 	u32 aen_result;
 	unsigned int shutdown_timeout;
 	unsigned int kato;
@@ -193,7 +192,6 @@ struct nvme_ctrl {
 	struct delayed_work ka_work;
 	struct nvme_command ka_cmd;
 	struct work_struct fw_act_work;
-	unsigned long events;
 
 #ifdef CONFIG_NVME_MULTIPATH
 	/* asymmetric namespace access: */
@@ -213,6 +211,10 @@ struct nvme_ctrl {
 	u16 maxcmd;
 	int nr_reconnects;
 	struct nvmf_ctrl_options *opts;
+#ifndef __GENKSYMS__
+	u32 oaes;
+	unsigned long events;
+#endif
 };
 
 #ifdef CONFIG_NVME_MULTIPATH
@@ -265,9 +265,11 @@ struct nvme_subsystem {
 	u8			cmic;
 	u16			vendor_id;
 	struct ida		ns_ida;
+#ifndef __GENKSYMS__
 #ifdef CONFIG_NVME_MULTIPATH
 	enum nvme_iopolicy	iopolicy;
 #endif
+#endif
 };
 
 /*
-- 
2.12.3

